ballerina/xmldata.bal

```diff
@@ -43,53 +43,63 @@ public type JsonOptions record {
 # successfully converted or else an `xmldata:Error`
 public isolated function fromJson(json jsonValue, JsonOptions options = {}) returns xml?|Error {
+    map<string> allNamespaces = {};
     if !isSingleNode(jsonValue) {
-        return getElement("root", check traverseNode(jsonValue, {}, options),
-                           check getAttributesMap(jsonValue, options = options));
+        addNamespaces(allNamespaces, check getNamespacesMap(jsonValue, {}, options));
+        return getElement("root", check traverseNode(jsonValue, allNamespaces, {}, options), allNamespaces,
+                            check getAttributesMap(jsonValue, allNamespaces, options = options));
     } else {
         map<json>|error jMap = jsonValue.ensureType();
         if jMap is map<json> {
             if jMap.length() == 0 {
                 return xml ``;
             }
             json value = jMap.toArray()[0];
+            addNamespaces(allNamespaces, check getNamespacesMap(value, {}, options));
             if value is json[] {
-                return getElement("root", check traverseNode(value, {}, options, jMap.keys()[0]),
-                                check getAttributesMap(value, options = options));
+                return getElement("root", check traverseNode(value, allNamespaces, {}, options, jMap.keys()[0]),
+                                allNamespaces, check getAttributesMap(value, allNamespaces, options = options));
             } else {
                 string key = jMap.keys()[0];
                 if key == CONTENT {
                     return xml:createText(value.toString());
                 }
-                return getElement(jMap.keys()[0], check traverseNode(value, {}, options),
-                                check getAttributesMap(value, options = options));
+                addNamespaces(allNamespaces, check getNamespacesMap(value, {}, options));
+                return getElement(jMap.keys()[0], check traverseNode(value, allNamespaces, {}, options), allNamespaces,
+                                check getAttributesMap(value, allNamespaces, options = options));
             }
         }
         if jsonValue !is null {
-           return xml:createText(jsonValue.toString());
+            return xml:createText(jsonValue.toString());
         } else {
-           return xml ``;
+            return xml ``;
         }
     }
 }
 
-isolated function traverseNode(json jNode, map<string> parentNamespaces, JsonOptions options = {},
-                                string? key = ()) returns xml|Error {
+isolated function traverseNode(json jNode, map<string> allNamespaces, map<string> parentNamespaces,
+                               JsonOptions options = {}, string? key = ()) returns xml|Error {
     string arrayEntryTag = options.arrayEntryTag == "" ? "item" : options.arrayEntryTag;
     string attributePrefix = options.attributePrefix == "" ? "@" : options.attributePrefix;
```

Comment:
why do we need to check whether `arrayEntryTag` and `attributePrefix` are empty? we already set default values for them right?